tests/unit/metadata/test_metadata_pkg_resources.py

```diff
@@ -82,7 +82,7 @@ def test_wheel_metadata_works() -> None:
     name = "simple"
     version = "0.1.0"
     require_a = "a==1.0"
-    require_b = 'b==1.1; extra == "also_b"'
+    require_b = 'b==1.1; extra == "also-b"'
```

Comment:
This needs investigation, I think.

Reply:
`iter_provided_extras` seems to be normalising things, which may be due to lack of round-tripping in `packaging` (https://github.com/pypa/packaging/issues/644#issuecomment-1626274566)?

Reply:
Oh, this is due to `_normalize_extra_values` in packaging! 

https://github.com/pypa/packaging/commit/e8a06183c1958dfc45be802b6192619497638842 is the relevant commit.